.github/workflows/docker-publish.yml

```diff
@@ -28,88 +34,37 @@ jobs:
-      - id: ecr_password
-        name: Retrieve ECR password
-        env:
-          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          AWS_DEFAULT_REGION: ap-southeast-1
-        run: echo "ecr_password=$(aws ecr get-login-password)" >> $GITHUB_OUTPUT
-      - id: image_tag
-        name: Setup image tags
-        run: |
-          IMAGE_TAG=latest
-          echo "image_tag=$IMAGE_TAG" >> $GITHUB_OUTPUT
-          echo "container_image_uri=$ECR_REGISTRY/$ECR_BUILD_REPO:$IMAGE_TAG" >> $GITHUB_OUTPUT
 
-  docker_build:
-    if: ${{ github.event_name == 'push' }}
-    timeout-minutes: 40
-    needs:
-      - setup_env
-    environment: test
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Check Out Repo
-        uses: actions/checkout@v3
-
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v2
-        with:
-          buildkitd-flags: --debug
-          install: true
-
-      - name: Cache Docker layers
-        uses: actions/cache@v3
-        with:
-          path: /tmp/.buildx-cache-${{ matrix.gemfile }}
-          key: ${{ runner.os }}-multi-buildx-${{ github.sha }}-${{ matrix.gemfile }}
-          restore-keys: |
-            ${{ runner.os }}-multi-buildx
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
 
-      - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ap-southeast-1
-
-      - name: Login to ECR
-        uses: docker/login-action@v2
-        with:
-          registry: ${{ env.ECR_REGISTRY }}
+    steps:
+      - uses: actions/checkout@v4
 
       - name: Build image
-        uses: docker/build-push-action@v3
-        env:
-          CONTAINER_IMAGE_URI: ${{ needs.setup_env.outputs.container_image_uri }}
-        with:
-          context: .
-          builder: ${{ steps.buildx.outputs.name }}
-          file: Dockerfile
-          push: true
-          tags: ${{ env.CONTAINER_IMAGE_URI }}
-          cache-from: type=local,src=/tmp/.buildx-cache-${{ matrix.gemfile }}
-          # Note the mode=max here
-          # More: https://github.com/moby/buildkit#--export-cache-options
-          # And: https://github.com/docker/buildx#--cache-tonametypetypekeyvalue
-          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new-${{ matrix.gemfile }}
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME
+
+      - name: Log into GitHub Container Registry
+        # TODO: Create a PAT with `read:packages` and `write:packages` scopes and save it as an Actions secret `CR_PAT`
+        run: echo "${{ secrets.CR_PAT }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
 
-      - name: Move cache
+      - name: Push image to GitHub Container Registry
```

Comment:
I think we can just used the `GITHUB_TOKEN` for login.
How about we use docker actions?
Eg:
```
      - name: Log in to the GitHub Container registry
        uses: docker/login-action@v3
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
      - name: Build and push Docker image
        id: push
        uses: docker/build-push-action@v5
        with:
          context: .
          push: true
          tags: $IMAGE_NAME
          labels: $IMAGE_NAME
```

Reply:
sure